error[E0308]: mismatched types
 --> $DIR/forget-semicolon.rs:8:5
  |
8 |     Some(()) //~ ERROR mismatched types
  |     ^^^^^^^^- help: try adding a semicolon: `;`
  |     |
  |     expected `()`, found enum `std::option::Option`
  |
  = note: expected unit type `()`
                  found enum `std::option::Option<()>`

error[E0308]: mismatched types
  --> $DIR/forget-semicolon.rs:14:5
   |
14 |     Some(()) //~ ERROR mismatched types
   |     ^^^^^^^^- help: try adding a semicolon: `;`
   |     |
   |     expected `()`, found enum `std::option::Option`
   |
   = note: expected unit type `()`
                   found enum `std::option::Option<()>`
