error: unexpected token
 --> $DIR/invalid-arguments.rs:8:16
  |
8 |     #[for_await(bar)] //~ ERROR unexpected token
  |                ^^^^^

error: expected `item`
  --> $DIR/invalid-arguments.rs:14:16
   |
14 | #[async_stream(baz, item = i32)] //~ ERROR expected `item`
   |                ^^^

error: unexpected token
  --> $DIR/invalid-arguments.rs:22:26
   |
22 | #[async_stream(item = i32, baz)] //~ ERROR unexpected token
   |                          ^

error: aborting due to 3 previous errors

