error: async stream must be declared as async
 --> $DIR/invalid.rs:7:11
  |
7 |     const fn constness() {} //~ ERROR async stream must be declared as async
  |           ^^

error: async stream must be declared as async
  --> $DIR/invalid.rs:10:5
   |
10 |     fn variadic(_: ...) {} //~ ERROR only foreign functions are allowed to be C-variadic
   |     ^^

error: async stream must be declared as async
  --> $DIR/invalid.rs:13:5
   |
13 |     fn asyncness() {} //~ ERROR async stream must be declared as async
   |     ^^

error: async stream must return the unit type
  --> $DIR/invalid.rs:16:26
   |
16 |     async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                          ^^^

error: duplicate #[stream] attribute
  --> $DIR/invalid.rs:29:5
   |
29 |     #[stream(item = ())] //~ ERROR duplicate #[stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> $DIR/invalid.rs:33:5
   |
33 |     #[try_stream(ok = (), error = ())] //~ ERROR duplicate #[try_stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
  --> $DIR/invalid.rs:37:5
   |
37 |     #[try_stream(ok = (), error = ())] //~ ERROR may not be used at the same time
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: duplicate #[stream] attribute

error: duplicate #[try_stream] attribute

error: duplicate #[for_await] attribute
  --> $DIR/invalid.rs:63:9
   |
63 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
   |         ^^^^^^^^^^^^
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: #[stream] and #[try_stream] may not be used at the same time

error: duplicate #[stream] attribute
  --> $DIR/invalid.rs:83:13
   |
83 |             #[stream] //~ ERROR duplicate #[stream] attribute
   |             ^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> $DIR/invalid.rs:92:13
   |
92 |             #[try_stream] //~ ERROR duplicate #[try_stream] attribute
   |             ^^^^^^^^^^^^^

error: duplicate #[for_await] attribute
   --> $DIR/invalid.rs:100:9
    |
100 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
    |         ^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
   --> $DIR/invalid.rs:108:13
    |
108 |             #[try_stream] //~ ERROR may not be used at the same time
    |             ^^^^^^^^^^^^^
