error: only foreign functions are allowed to be C-variadic
  --> $DIR/invalid-function.rs:12:16
   |
12 | fn variadic(_: ...) {} //~ ERROR async stream may not be variadic
   |                ^^^

error: async stream may not be const
 --> $DIR/invalid-function.rs:9:1
  |
9 | const fn constness() {} //~ ERROR async stream may not be const
  | ^^^^^

error: expected `:`
  --> $DIR/invalid-function.rs:11:1
   |
11 | #[async_stream(item = ())]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: async stream must be declared as async
  --> $DIR/invalid-function.rs:15:1
   |
15 | fn asyncness() {} //~ ERROR async stream must be declared as async
   | ^^

error: async stream must return the unit type
  --> $DIR/invalid-function.rs:18:22
   |
18 | async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                      ^^^

error: aborting due to 5 previous errors

