error: unexpected token: (bar)
  --> $DIR/invalid-arguments.rs:14:16
   |
14 |     #[for_await(bar)] //~ ERROR unexpected token
   |                ^^^^^

error: expected `item`
  --> $DIR/invalid-arguments.rs:20:16
   |
20 | #[async_stream(baz, item = i32)] //~ ERROR expected `item`
   |                ^^^

error: expected `item`
  --> $DIR/invalid-arguments.rs:25:28
   |
25 | #[async_stream(item = i32, baz)] //~ ERROR unexpected token
   |                            ^^^

error: expected `,`
  --> $DIR/invalid-arguments.rs:30:27
   |
30 | #[async_stream(item = i32 item = i32)] //~ ERROR expected `,`
   |                           ^^^^

error: duplicate `item` argument
  --> $DIR/invalid-arguments.rs:35:35
   |
35 | #[async_stream(item = i32, item = i32)] //~ ERROR duplicate `item` argument
   |                                   ^^^

error: duplicate `boxed` argument
  --> $DIR/invalid-arguments.rs:40:35
   |
40 | #[async_stream(item = i32, boxed, boxed)] //~ ERROR duplicate `boxed` argument
   |                                   ^^^^^

error: duplicate `boxed_local` argument
  --> $DIR/invalid-arguments.rs:45:41
   |
45 | #[async_stream(item = i32, boxed_local, boxed_local)] //~ ERROR duplicate `boxed_local` argument
   |                                         ^^^^^^^^^^^

error: `boxed` and `boxed_local` cannot be used at the same time.
  --> $DIR/invalid-arguments.rs:50:41
   |
50 | #[async_stream(item = i32, boxed_local, boxed)] //~ ERROR `boxed` and `boxed_local` cannot be used at the same time.
   |                                         ^^^^^
