error: only foreign functions are allowed to be C-variadic
  --> $DIR/invalid-function.rs:11:16
   |
11 | fn variadic(_: ...) {} //~ ERROR async stream may not be variadic
   |                ^^^

error: async stream must be declared as async
 --> $DIR/invalid-function.rs:8:7
  |
8 | const fn constness() {} //~ ERROR async stream must be declared as async
  |       ^^

error: expected `:`
  --> $DIR/invalid-function.rs:10:1
   |
10 | #[async_stream(item = ())]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: async stream must be declared as async
  --> $DIR/invalid-function.rs:14:1
   |
14 | fn asyncness() {} //~ ERROR async stream must be declared as async
   | ^^

error: async stream must return the unit type
  --> $DIR/invalid-function.rs:17:22
   |
17 | async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                      ^^^

error: aborting due to 5 previous errors

