error: async stream must be declared as async
 --> $DIR/invalid-function.rs:6:7
  |
6 | const fn constness() {} //~ ERROR async stream must be declared as async
  |       ^^

error: async stream must be declared as async
 --> $DIR/invalid-function.rs:9:1
  |
9 | fn variadic(_: ...) {} //~ ERROR only foreign functions are allowed to be C-variadic
  | ^^

error: async stream must be declared as async
  --> $DIR/invalid-function.rs:12:1
   |
12 | fn asyncness() {} //~ ERROR async stream must be declared as async
   | ^^

error: async stream must return the unit type
  --> $DIR/invalid-function.rs:15:22
   |
15 | async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                      ^^^
