error: unexpected token
  --> $DIR/invalid-arguments.rs:16:16
   |
16 |     #[for_await(bar)] //~ ERROR unexpected token
   |                ^

error: expected `item`
  --> $DIR/invalid-arguments.rs:22:16
   |
22 | #[async_stream(baz, item = i32)] //~ ERROR expected `item`
   |                ^^^

error: expected `item`
  --> $DIR/invalid-arguments.rs:27:28
   |
27 | #[async_stream(item = i32, baz)] //~ ERROR unexpected token
   |                            ^^^

error: expected `,`
  --> $DIR/invalid-arguments.rs:32:27
   |
32 | #[async_stream(item = i32 item = i32)] //~ ERROR expected `,`
   |                           ^^^^

error: duplicate `item` argument
  --> $DIR/invalid-arguments.rs:37:35
   |
37 | #[async_stream(item = i32, item = i32)] //~ ERROR duplicate `item` argument
   |                                   ^^^

error: duplicate `boxed` argument
  --> $DIR/invalid-arguments.rs:42:35
   |
42 | #[async_stream(item = i32, boxed, boxed)] //~ ERROR duplicate `boxed` argument
   |                                   ^^^^^

error: duplicate `boxed_local` argument
  --> $DIR/invalid-arguments.rs:47:41
   |
47 | #[async_stream(item = i32, boxed_local, boxed_local)] //~ ERROR duplicate `boxed_local` argument
   |                                         ^^^^^^^^^^^

error: `boxed` and `boxed_local` cannot be used at the same time.
  --> $DIR/invalid-arguments.rs:52:41
   |
52 | #[async_stream(item = i32, boxed_local, boxed)] //~ ERROR `boxed` and `boxed_local` cannot be used at the same time.
   |                                         ^^^^^
