error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async fn` body must be known in this context
  --> $DIR/question-mark-await-type-error.rs:13:15
   |
13 |     for _i in 1..2 {
   |               ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async fn` body because of this `await`
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async fn` body must be known in this context
  --> $DIR/question-mark-await-type-error.rs:13:18
   |
13 |     for _i in 1..2 {
   |                  ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async fn` body because of this `await`
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async fn` body must be known in this context
  --> $DIR/question-mark-await-type-error.rs:13:15
   |
13 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async fn` body because of this `await`
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async fn` body must be known in this context
  --> $DIR/question-mark-await-type-error.rs:13:9
   |
13 |     for _i in 1..2 {
   |         ^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async fn` body because of this `await`
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:27:5
   |
27 |     #[for_await]
   |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:21:18
   |
21 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |                  ^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside generator must be known in this context
  --> $DIR/question-mark-await-type-error.rs:20:15
   |
20 |     for _i in 1..2 {
   |               ^ cannot infer type for `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/question-mark-await-type-error.rs:21:18
   |
21 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |                  ^^^^^

error[E0698]: type inside generator must be known in this context
  --> $DIR/question-mark-await-type-error.rs:20:18
   |
20 |     for _i in 1..2 {
   |                  ^ cannot infer type for `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/question-mark-await-type-error.rs:21:18
   |
21 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |                  ^^^^^

error[E0698]: type inside generator must be known in this context
  --> $DIR/question-mark-await-type-error.rs:20:15
   |
20 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/question-mark-await-type-error.rs:21:18
   |
21 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |                  ^^^^^

error[E0698]: type inside generator must be known in this context
  --> $DIR/question-mark-await-type-error.rs:20:9
   |
20 |     for _i in 1..2 {
   |         ^^ cannot infer type for `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/question-mark-await-type-error.rs:21:18
   |
21 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |                  ^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:40:18
   |
40 |         async {}.await?;
   |                  ^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`
