error: cannot find attribute macro `async_stream` in this scope
  --> $DIR/question-mark-await-type-error.rs:12:3
   |
12 | #[async_stream(item = i32)]
   |   ^^^^^^^^^^^^

error: cannot find attribute macro `async_stream` in this scope
  --> $DIR/question-mark-await-type-error.rs:29:3
   |
29 | #[async_stream(item = i32)]
   |   ^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
  |
  = help: the trait `std::ops::Try` is not implemented for `()`
  = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
 --> $DIR/question-mark-await-type-error.rs:7:15
  |
7 |     for _i in 1..2 {
  |               ^^^^ cannot infer type for `{integer}`
  |
note: the type is part of the `async` object because of this `await`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
 --> $DIR/question-mark-await-type-error.rs:7:15
  |
7 |     for _i in 1..2 {
  |               ^ cannot infer type for `{integer}`
  |
note: the type is part of the `async` object because of this `await`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
 --> $DIR/question-mark-await-type-error.rs:7:18
  |
7 |     for _i in 1..2 {
  |                  ^ cannot infer type for `{integer}`
  |
note: the type is part of the `async` object because of this `await`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
 --> $DIR/question-mark-await-type-error.rs:7:15
  |
7 |     for _i in 1..2 {
  |               ^^^^ cannot infer type for `{integer}`
  |
note: the type is part of the `async` object because of this `await`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
 --> $DIR/question-mark-await-type-error.rs:7:9
  |
7 |     for _i in 1..2 {
  |         ^^ cannot infer type for `{integer}`
  |
note: the type is part of the `async` object because of this `await`
 --> $DIR/question-mark-await-type-error.rs:8:9
  |
8 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
  |         ^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:21:5
   |
21 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:14:15
   |
14 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:14:15
   |
14 |     for _i in 1..2 {
   |               ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:14:18
   |
14 |     for _i in 1..2 {
   |                  ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:14:15
   |
14 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:14:9
   |
14 |     for _i in 1..2 {
   |         ^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:15:9
   |
15 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:31:5
   |
31 |     #[for_await]
   |     ^^^^^^^^^^^^

error: aborting due to 22 previous errors

Some errors have detailed explanations: E0277, E0698.
For more information about an error, try `rustc --explain E0277`.
