error: cannot find attribute macro `async_stream` in this scope
  --> $DIR/question-mark-await-type-error.rs:15:3
   |
15 | #[async_stream(item = i32)]
   |   ^^^^^^^^^^^^

error: cannot find attribute macro `async_stream` in this scope
  --> $DIR/question-mark-await-type-error.rs:32:3
   |
32 | #[async_stream(item = i32)]
   |   ^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:10:15
   |
10 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:10:15
   |
10 |     for _i in 1..2 {
   |               ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:10:18
   |
10 |     for _i in 1..2 {
   |                  ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:10:15
   |
10 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:10:9
   |
10 |     for _i in 1..2 {
   |         ^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:11:9
   |
11 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:24:5
   |
24 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:17:15
   |
17 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:17:15
   |
17 |     for _i in 1..2 {
   |               ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:17:18
   |
17 |     for _i in 1..2 {
   |                  ^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:17:15
   |
17 |     for _i in 1..2 {
   |               ^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:17:9
   |
17 |     for _i in 1..2 {
   |         ^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:18:9
   |
18 |         async {}.await?; //~ ERROR the `?` operator can only be applied to values that implement `std::ops::Try`
   |         ^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Try`
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Try` is not implemented for `()`
   = note: required by `std::ops::Try::into_result`

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^

error[E0698]: type inside `async` object must be known in this context
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^ cannot infer type for `{integer}`
   |
note: the type is part of the `async` object because of this `await`
  --> $DIR/question-mark-await-type-error.rs:34:5
   |
34 |     #[for_await]
   |     ^^^^^^^^^^^^

error: aborting due to 22 previous errors

Some errors have detailed explanations: E0277, E0698.
For more information about an error, try `rustc --explain E0277`.
